### ISSUE TITLE
Sox 14.4.2 Failing to compile with 64 bit

### ISSUE THREAD

**Issue by randomnicode** [m-ab-s/media-autobuild_suite#1426]
libsox 14.4.2 failing to compile for a 64-bit build. Have not tested for 32 bit build. Log attached

``` bash
sox 14.4.2  .............................................. [Not installed]
├ sox-14.4.2  .......................................... [File up-to-date]
Installing libmad... done
Installing opusfile... done
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
F:/dev/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ./.libs/libsox.a(libsox_la-flac.o):flac.c:(.text+0xb77): undefined reference to `__imp_FLAC__stream_decoder_process_until_end_of_metadata'
F:/dev/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ./.libs/libsox.a(libsox_la-flac.o):flac.c:(.text+0xb85): undefined reference to `__imp_FLAC__stream_decoder_get_state'
F:/dev/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ./.libs/libsox.a(libsox_la-flac.o):flac.c:(.text+0xcf9): undefined reference to `__imp_FLAC__stream_encoder_get_state'
F:/dev/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ./.libs/libsox.a(libsox_la-flac.o):flac.c:(.text+0xd05): undefined reference to `__imp_FLAC__stream_encoder_finish'
F:/dev/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ./.libs/libsox.a(libsox_la-flac.o):flac.c:(.text+0xd0f): undefined reference to `__imp_FLAC__stream_encoder_delete'
F:/dev/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ./.libs/libsox.a(libsox_la-flac.o):flac.c:(.text+0xd21): undefined reference to `__imp_FLAC__metadata_object_delete'
collect2.exe: error: ld returned 1 exit status
make[1]: *** [Makefile:1410: sox.exe] Error 1
make[1]: Leaving directory '/build/sox-14.4.2/build-64bit/src'
make: *** [Makefile:644: all-recursive] Error 1
make failed. Check F:/dev/media-autobuild_suite/build/sox-14.4.2/build-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

```

[log.zip](https://github.com/m-ab-s/media-autobuild_suite/files/3718756/zxXv.zip)

**Comment by 1480c1** [comment]
https://github.com/xiph/flac/commit/b917d456d23c3268cc6f466d720605b713d7b9ff

Seems they made it so that flac will use shared prefixed functions

**Comment by wiiaboo** [comment]
Well, easily fixed. Just copy the other dllexport/import seds that are already in stuff like sdl2.

**Comment by randomnicode** [comment]
Sorry to reopen the issue again, but that did not fix it. Even after recompiling flac, I'm still running into the same issue
```
D:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ./.libs/libsox.a(libsox_la-flac.o):flac.c:(.text+0xd05): undefined reference to `__imp_FLAC__stream_encoder_finish'
D:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ./.libs/libsox.a(libsox_la-flac.o):flac.c:(.text+0xd0f): undefined reference to `__imp_FLAC__stream_encoder_delete'
D:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ./.libs/libsox.a(libsox_la-flac.o):flac.c:(.text+0xd21): undefined reference to `__imp_FLAC__metadata_object_delete'
collect2.exe: error: ld returned 1 exit status
```

**Comment by 1480c1** [comment]
Right now testing a fix

**Comment by nikolasr** [comment]
An ugly really quick'n'dirty workaround would be to edit media-suite_compile.sh so that line 585 says
`if [[ $flac = y ]] && do_vcs "https://github.com/xiph/flac.git#tag=2e7931c27eb15e387da440a37f12437e35b22dd4"; then`.
However this would mean that the flac git source is pinpointed to tree 2e7931c27eb15e387da440a37f12437e35b22dd4 and no newer versions will be downloaded. So this needs to be undone when a true fix is available.

**Comment by mustafahlvc** [comment]
Hi @nikolasr the workaround didn't work for me.

`Likely error (tail of the failed operation logfile):
C:/mingw/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: flac.o:flac.c:(.text+0x6d8): undefined reference to `__imp_FLAC__stream_decoder_finish'
C:/mingw/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: flac.o:flac.c:(.text+0x6e2): undefined reference to `__imp_FLAC__stream_decoder_delete'
collect2.exe: error: ld returned 1 exit status
make[3]: *** [Makefile:540: oggenc.exe] Error 1
make[3]: Leaving directory '/build/vorbis-tools-git/build-64bit/oggenc'
make[2]: *** [Makefile:598: all-recursive] Error 1
make[2]: Leaving directory '/build/vorbis-tools-git/build-64bit/oggenc'
make[1]: *** [Makefile:497: all-recursive] Error 1
make[1]: Leaving directory '/build/vorbis-tools-git/build-64bit'
make: *** [Makefile:427: all] Error 2
make failed. Check C:/mingw/build/vorbis-tools-git/build-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
`

**Comment by nikolasr** [comment]
Well, OK. Let's try to work that out ;-)

This is what I did: 
I modified media-suite_compile.sh. I used Notepad++ for this. (Beware of line change options for Windows and Linux when using other editors!)
Watch out fot that line change... it's _one_ whole line.
I deleted media-autobuild_suite-master\build\flac-git in my directory.
Just search for flac-git and delete that dir.
To be sure it gets downloaded again.
I restarted media-autobuild_suite.bat.
And now the installer said for flac "recently updated", and compiled. If it doesn't You'll have a problem ... :-(

So I can imagine 2 things: 
- media-suite_compile.sh not properly modified (most likely!)
- old config files (try a clean build)

HTH, Nick!
